### PR TITLE
INTYGFV-14488: Mapping from internal to certificate for TsBasV6.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**
WIP

### CODE REVIEW

**Review by mikeyOne — CHANGES_REQUESTED** [review]
Looks good, just a few comments to consider...

File: support/src/testFixtures/java/se/inera/intyg/common/support/facade/testsetup/model/config/ConfigTest.java [hunk 1]
```diff
@@ -90,8 +90,6 @@ void shouldIncludeConfigText() {
         if (getTextProviderMock() != null && getTextId() != null && !question.getConfig().getText().equals(getTextId())) {
             assertTrue(question.getConfig().getText().trim().length() > 0, "Missing text");
             verify(getTextProviderMock(), atLeastOnce()).get(getTextId());
-        } else {
-            assertEquals(getTextId(), question.getConfig().getText());
```
[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply support/src/testFixtures/java/se/inera/intyg/common/support/facade/testsetup/model/config/ConfigTest.java:94]: Was lost during resolving of merge conflicts, its now back again!

File: ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v6/model/converter/certificate/question/QuestionIdentitet.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                return IdKontrollKod.ID_KORT.getDescription();
+            case "FORETAG_ELLER_TJANSTEKORT":
+                return IdKontrollKod.FORETAG_ELLER_TJANSTEKORT.getDescription();
+            case "KORKORT":
+                return IdKontrollKod.KORKORT.getDescription();
+            case "PERS_KANNEDOM":
+                return IdKontrollKod.PERS_KANNEDOM.getDescription();
+            case "FORSAKRAN_KAP18":
+                return IdKontrollKod.FORSAKRAN_KAP18.getDescription();
+            case "PASS":
```
[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v6/model/converter/certificate/question/QuestionIdentitet.java:70]: Not to my knowledge. The enum only has a code-value that we can map against.

image

[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v6/model/converter/certificate/question/QuestionIdentitet.java:70]: Changed the hard coded values to static final Strings instead.

image

File: ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v6/model/converter/certificate/question/QuestionUppfattaSamtal4Meter.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import static se.inera.intyg.common.ts_bas.v6.codes.RespConstantsV6.UPPFATTA_SAMTALSTAMMA_SVAR_ID;
+import static se.inera.intyg.common.ts_bas.v6.codes.RespConstantsV6.UPPFATTA_SAMTALSTAMMA_SVAR_TEXT_ID;
+
+import se.inera.intyg.common.services.texts.CertificateTextProvider;
+import se.inera.intyg.common.support.facade.model.CertificateDataElement;
+import se.inera.intyg.common.support.facade.model.config.CertificateDataConfigViewText;
+import se.inera.intyg.common.support.facade.model.value.CertificateDataValueViewText;
+import se.inera.intyg.common.ts_bas.v6.model.internal.HorselBalans;
+
+public class QuestionUppfattaSamtal4Meter {
```
[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v6/model/converter/certificate/question/QuestionUppfattaSamtal4Meter.java:33]: I agree. I have taken inspiration for this nameing of this question from TsBasV7. I will change it for both versions.

File: ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v6/codes/RespConstantsV6.java [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.inera.intyg.common.ts_bas.v6.codes;
+
+public final class RespConstantsV6 {
```
[the remark this answers is not in the document]
Comment (mikeyOne) [reply ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v6/codes/RespConstantsV6.java:21]: Yes, that's what I was thinking too :)

**Review by mikeyOne — APPROVED** [review]

**Pull request merged**